File: server/app/schemas/grocery.py
from __future__ import annotations
from pydantic import BaseModel
from datetime import date


class GroceryLineItem(BaseModel):
    ingredient_id: int
    ingredient_name: str
    ingredient_category: str
    combined_quantity: str
    unit: str
    source_recipes: list[str]


class GroceryListResponse(BaseModel):
    week_plan_id: int
    week_start: date
    items: list[GroceryLineItem]
    grouped_by_category: dict[str, list[GroceryLineItem]]

File: apps/meal-planner/src/types/grocery.ts
export interface GroceryLineItem {
  ingredient_id: number
  ingredient_name: string
  ingredient_category: string
  combined_quantity: string
  unit: string
  source_recipes: string[]
}

export interface GroceryListResponse {
  week_plan_id: number
  week_start: string
  items: GroceryLineItem[]
  grouped_by_category: Record<string, GroceryLineItem[]>
}
